crates/viewer/re_space_view_dataframe/src/dataframe_ui.rs

```diff
@@ -0,0 +1,350 @@
+use std::collections::BTreeMap;
+use std::ops::Range;
+
+use anyhow::Context;
+use itertools::Itertools;
+
+use re_chunk_store::{ColumnDescriptor, LatestAtQuery, RowId};
+use re_dataframe::{LatestAtQueryHandle, RangeQueryHandle, RecordBatch};
+use re_log_types::{EntityPath, TimeInt, Timeline};
+use re_types_core::ComponentName;
+use re_ui::UiExt as _;
+use re_viewer_context::ViewerContext;
+
+use crate::display_record_batch::{DisplayRecordBatch, DisplayRecordBatchError};
+
+/// Display a dataframe table for the provided query.
+pub(crate) fn dataframe_ui<'a>(
+    ctx: &ViewerContext<'_>,
+    ui: &mut egui::Ui,
+    query: impl Into<QueryHandle<'a>>,
+) {
+    dataframe_ui_impl(ctx, ui, &query.into());
+}
+
+/// A query handle for either a latest-at or range query.
+pub(crate) enum QueryHandle<'a> {
+    LatestAt(LatestAtQueryHandle<'a>),
+    Range(RangeQueryHandle<'a>),
+}
+
+impl QueryHandle<'_> {
+    fn schema(&self) -> &[ColumnDescriptor] {
+        match self {
+            QueryHandle::LatestAt(query_handle) => query_handle.schema(),
+            QueryHandle::Range(query_handle) => query_handle.schema(),
+        }
+    }
+
+    fn num_rows(&self) -> u64 {
+        match self {
+            QueryHandle::LatestAt(_) => 1,
+            QueryHandle::Range(query_handle) => query_handle.num_rows(),
+        }
+    }
+
+    fn get(&self, start: u64, num_rows: u64) -> Vec<RecordBatch> {
+        match self {
+            QueryHandle::LatestAt(query_handle) => {
+                vec![query_handle.get()]
```

Comment:
Is it intentional that the row range is ignored? Should this be documented? Should we have a `debug_assert` to check that the user asked for `0..1`?